diff --git a/src/algorithms/convdirect_block_blis.c b/src/algorithms/convdirect_block_blis.c
index a4de816..4d9b35b 100644
--- a/src/algorithms/convdirect_block_blis.c
+++ b/src/algorithms/convdirect_block_blis.c
@@ -173,11 +173,12 @@ void CONVDIRECT_KERNEL_WITH_PARAMS {
                         int x = vstride * l + vdilation * n - vpadding;
                         if (x < 0 || x > Ho) continue; // skip the whole thing
                         for (m = 0; m < Wf; m++) {
+#ifdef ARCH__aarch64__
                             packRB_neon('R', 'N', kb, ib, &Drow_NHWC(h, i, l + n, k + m), ldD3, Ac, MR);
+#else
                             // assert(i + ib <= Ci); assert(k + kb <= wo); // assert(k + kb + m <= Wo);
-                //packRB('R', 'N', kb, ib, D + h * ldD1 + x * ldD2 + i,
-                            //k, m, Wo, hpadding, hstride, hdilation,
-                            //ldD3, Ac, MR);
+                            packRB('R', 'N', kb, ib, D + h * ldD1 + x * ldD2 + i, k, m, Wo, hpadding, hstride, hdilation, ldD3, Ac, MR);
+#endif
                 //j2 = 0;
                             for (j = 0, j2 = 0; j < Co; j += COB, j2++) {
                             //for (j = 0; j < Co; j += COB) {
diff --git a/src/algorithms/packrb.c b/src/algorithms/packrb.c
index 0cf3d12..efa3530 100644
--- a/src/algorithms/packrb.c
+++ b/src/algorithms/packrb.c
@@ -25,6 +25,7 @@
 
 #define Acol(a1,a2)  A[ (a2)*(ldA)+(a1) ]
 
+#ifdef ARCH__aarch64__
 void packRB_neon( char orderA, char transA, int mc, int nc, DTYPE *A, int ldA, DTYPE *Ac, int RR ) {
   
     int    i, j, ii, ii_tmp, k, rr, iters, iters_left;
@@ -188,6 +189,7 @@ void packRB_neon( char orderA, char transA, int mc, int nc, DTYPE *A, int ldA, D
     }
 
 }
+#endif
 
 void packRB(char orderA, char transA, int mc, int nc, const DTYPE *A,
         int start_y, int ky, int dim_w,
diff --git a/src/algorithms/packrb.h b/src/algorithms/packrb.h
index 5006fe7..a357a3d 100644
--- a/src/algorithms/packrb.h
+++ b/src/algorithms/packrb.h
@@ -20,13 +20,17 @@
 #include "../dtypes.h"
 #include "../arrays.h"
 #include "../macros.h"
+#ifdef ARCH__aarch64__
 #include <arm_neon.h>
+#endif
 
 #ifndef CONVDIRECT_PACKRB_H
 #define CONVDIRECT_PACKRB_H
 
 
+#ifdef ARCH__aarch64__
 void packRB_neon( char orderA, char transA, int mc, int nc, DTYPE *A, int ldA, DTYPE *Ac, int RR );
+#endif
 
 void packRB(char orderA, char transA, int mc, int nc, const DTYPE *A,
         int start_y, int ky, int dim_w,
